packages/resolve-scripts/src/runtime/server/get_html_markup.js

```diff
@@ -0,0 +1,29 @@
+import { Helmet } from 'react-helmet'
+
+import jsonUtfStringify from './utils/json_utf_stringify'
+
+export default ({ markup, styleTags, initialState, env, clientUrl }) => {
+  const helmet = Helmet.renderStatic()
+
+  return (
+    `<!doctype html>` +
+    `<html ${helmet.htmlAttributes.toString()}>` +
+    '<head>' +
+    `${helmet.title.toString()}` +
+    `${helmet.meta.toString()}` +
+    `${helmet.link.toString()}` +
+    `${helmet.style.toString()}` +
+    styleTags +
+    '<script>' +
+    `window.__INITIAL_STATE__=${jsonUtfStringify(initialState)};` +
```

Comment:
Also as comment above - even if provide initial state to application, no view model can use it by current architecture. From another side, it can be used by custom UI reducers, e.g. for provide optimistic page loading... It's discussion queston